components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -157,6 +181,20 @@ void ViewCounterService::Shutdown() {
 }
 
 void ViewCounterService::OnUpdated(NTPBackgroundImagesData* data) {
+  if (data != GetCurrentWallpaperData())
+    return;
+
+  DVLOG(2) << __func__ << ": Active data is updated.";
+
+  // Data is updated, so change our stored data and reset any indexes.
+  // But keep view counter until branded content is seen.
+  if (data) {
+    model_.ResetCurrentWallpaperImageIndex();
```

Comment:
Why current branded wallpaper index is also reset together?
This callback is called when NTP BI component is updated.
So, I think only `current_wallpaper_image_index_` should be reset.

Reply:
and above comment seems to be updated.